src/catalog/src/information_schema/region_peers.rs

```diff
@@ -0,0 +1,275 @@
+                builder
+                    .make_region_peers(Some(request))
+                    .await
+                    .map(|x| x.into_df_record_batch())
+                    .map_err(Into::into)
+            }),
+        ));
+        Ok(Box::pin(
+            RecordBatchStreamAdapter::try_new(stream)
+                .map_err(BoxedError::new)
+                .context(InternalSnafu)?,
+        ))
+    }
+}
+
+struct InformationSchemaRegionPeersBuilder {
+    schema: SchemaRef,
+    catalog_name: String,
+    catalog_manager: Weak<dyn CatalogManager>,
+
+    region_ids: UInt64VectorBuilder,
+    peer_ids: UInt64VectorBuilder,
+    peer_addrs: StringVectorBuilder,
+    is_leaders: StringVectorBuilder,
+    statuses: StringVectorBuilder,
+    down_seconds: Int64VectorBuilder,
+}
+
+impl InformationSchemaRegionPeersBuilder {
+    fn new(
+        schema: SchemaRef,
+        catalog_name: String,
+        catalog_manager: Weak<dyn CatalogManager>,
+    ) -> Self {
+        Self {
+            schema,
+            catalog_name,
+            catalog_manager,
+            region_ids: UInt64VectorBuilder::with_capacity(42),
+            peer_ids: UInt64VectorBuilder::with_capacity(42),
+            peer_addrs: StringVectorBuilder::with_capacity(42),
+            is_leaders: StringVectorBuilder::with_capacity(42),
+            statuses: StringVectorBuilder::with_capacity(42),
+            down_seconds: Int64VectorBuilder::with_capacity(42),
+        }
+    }
+
+    /// Construct the `information_schema.region_peers` virtual table
+    async fn make_region_peers(&mut self, request: Option<ScanRequest>) -> Result<RecordBatch> {
+        let catalog_name = self.catalog_name.clone();
+        let catalog_manager = self
+            .catalog_manager
+            .upgrade()
+            .context(UpgradeWeakCatalogManagerRefSnafu)?;
+
+        let partition_manager = catalog_manager
+            .as_any()
+            .downcast_ref::<KvBackendCatalogManager>()
+            .map(|catalog_manager| catalog_manager.partition_manager());
+
+        let predicates = Predicates::from_scan_request(&request);
+
+        for schema_name in catalog_manager.schema_names(&catalog_name).await? {
+            if !catalog_manager
+                .schema_exists(&catalog_name, &schema_name)
+                .await?
+            {
+                continue;
+            }
+
+            let mut stream = catalog_manager.tables(&catalog_name, &schema_name).await;
+
+            while let Some(table) = stream.try_next().await? {
+                let table_info = table.table_info();
+
+                if table_info.table_type == TableType::Temporary {
+                    continue;
+                }
+
+                let table_id = table_info.ident.table_id;
+                let routes = if let Some(partition_manager) = &partition_manager {
+                    partition_manager
+                        .find_region_routes(table_id)
```

Comment:
Calling `find_region_routes` in a loop is expected to be very slow. It is recommended to split the steam and then get region routes in batches. Fortunately, we have already provided the [batch get](https://github.com/GreptimeTeam/greptimedb/blob/37e9f15b7715058f8676ffe02f4e03d2534ddf52/src/common/meta/src/key/table_route.rs#L392) method.

Reply:
We used a similar approach when scanning all [tables](https://github.com/GreptimeTeam/greptimedb/blob/de13de1454736253ae7725a4863452d6f0c6a18e/src/catalog/src/kvbackend/manager.rs#L266) information.